### PR TITLE
build(docker): ease local development

### PR DESCRIPTION
**Pull request by annibalsilva**
# Description

Try to ease local development with Docker. Changes include:

1. Allow setting JVM options via the `JAVA_OPTS` environment variable when running the application with Docker
2. Allow the optional use of a file with environment variables to be used in the scripts
3. Update the Compose file, using volumes so that we don't need to rebuild the image every time the application's source code changes

# How Has This Been Tested?

- [ ] Checking if the regular deployment (one which doesn't create a volume to the project's target folder) still works.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### CODE REVIEW

**Review by RMCampos — CHANGES_REQUESTED** [review]
Works great!! Good job. Just a small fix to keep it working with docker-compose.

File: dockerfile-entrypoint.sh [hunk 1]
```diff
@@ -6,5 +6,6 @@ keytool -importcert -alias orakey -noprompt -cacerts -storepass changeit -file o
 
 java \
     -Djava.security.egd=file:/dev/./urandom \
+    ${JAVA_OPTS} \
     -jar \
-    /usr/share/service/service.jar
+    /usr/share/service/target/backend-start-api.jar
```
[the remark this answers is not in the document]
Comment (annibalsilva) [reply dockerfile-entrypoint.sh]: I made some changes that work without volumes now.

**Review by DerekRoberts — APPROVED** [review]
Looks good!  :)

**Comment by annibalsilva** [comment]
> ...a few required jobs haven't completed. That's outside the scope of this PR, right?

Yes, this one is just to make local development with Docker easier.

**Review by RMCampos — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -2,8 +2,10 @@ version: '3.9'
 
 services:
   backend:
-    container_name: service-api
-    image: service-api:latest
+    build:
+      context: .
+    container_name: backend
+    dns: ${DNS}
```
[the remark this answers is not in the document]
Comment (annibalsilva) [reply docker-compose.yml:8]: Is it a good idea to publish the VPN's DNS address in the repository?

[the remark this answers is not in the document]
Comment (RMCampos) [reply docker-compose.yml:8]: Maybe we could give the command to populate this variable. nslookup.

**Review by RMCampos — APPROVED** [review]
LGTM

**Pull request merged**